Px.Utils/Validation/DataValidation/DataValidation.cs

```diff
@@ -0,0 +1,309 @@
+        /// <param name="streamEncoding">The encoding of the stream.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>An asynchronous enumerable of tokens.</returns>
+        public static async IAsyncEnumerable<Token> TokenizeAsync(Stream stream, PxFileSyntaxConf conf,
+            Encoding? streamEncoding, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+        {
+            const int streamBufferSize = 1024;
+            var charPosition = 0;
+            var lineNumber = 1;
+            using StreamReader reader = new(stream, streamEncoding, false, streamBufferSize, true);
+            var valueBuilder = new StringBuilder();
+            var buffer = new char[streamBufferSize];
+            var readStartIndex = 0;
+
+            var charsRead = 0;
+            var ignoreNext = false;
+            Token? token;
+            do
+            {
+                charsRead = await reader.ReadAsync(buffer.AsMemory(readStartIndex), cancellationToken);
+
+                for (var currentPos = 0; currentPos < charsRead - 1 + readStartIndex; currentPos++)
+                {
+                    if (ignoreNext)
+                    {
+                        ignoreNext = false;
+                        continue;
+                    }
+                    var currentCharacter = buffer[currentPos];
+                    var nextCharacter = buffer[currentPos + 1];
+
+                    token = GetToken(currentCharacter, nextCharacter, ref lineNumber, ref charPosition,
+                        out var skipNext,
+                        conf, ref valueBuilder);
+                    ignoreNext = skipNext;
+                    if (token != null)
+                    {
+                        yield return (Token)token;
+                    }
+                }
+                buffer[0] = buffer[charsRead - 1 + readStartIndex];
+                readStartIndex = 1;
+
+            } while (charsRead > 0);
+
+            if (!ignoreNext)
+            {
+                token = GetToken(buffer[0], 0, ref lineNumber, ref charPosition, out _, conf, ref valueBuilder);
+
+                if (token != null)
+                {
+                    yield return (Token)token;
+                }
+            }
+
+            yield return new Token(TokenType.EndOfStream, valueBuilder.ToString(), lineNumber, charPosition);
+        }
+
+
+        private static Token? GetToken(char currentCharacter, int nextCharacter, ref int lineNumber,
+            ref int charPosition, out bool skipNext, PxFileSyntaxConf conf, ref StringBuilder valueBuilder)
+        {
+            skipNext = false;
+            charPosition++;
+            if (currentCharacter is ' ' or '\t')
+            {
+                return new Token(TokenType.DataItemSeparator, currentCharacter.ToString(), lineNumber,
+                    charPosition);
+            } 
+            if (currentCharacter is '\n' or '\r')
```

Comment:
These hardcoded characters are defined in PxFile/CharacterConstants class (LINE_FEED, HORIZONTAL_TAB and CARRIAGE_RETURN if I recall). Those should be used where possible.

Reply:
Other such non-syntax defining (for which we have PxSyntaxConf) can be defined there too if missing and there's need.

Reply:
I can not see the benefit of definig constant for single characters if there is no other semantics involved. '\n' and ' ' are as readable as LINE_FEED and SPACE